src/yt_idefix/io.py

```diff
@@ -109,7 +109,7 @@ def _read_fluid_selection(self, chunks, selector, fields, size):
             2 4.998045e+00 3.400969e-03 1458 single_file little rho vx1 vx2 vx3 prs tr1 tr2 tr3 Temp ndens PbykB mach
             3 7.497932e+00 3.386245e-03 2186 single_file little rho vx1 vx2 vx3 prs tr1 tr2 tr3 Temp ndens PbykB mach
         """
-        if (match := re.search(r"\d{4}", self.ds.filename)) is not None:
+        if (match := re.search(r"\d{4}(?!.*\d{4})", self.ds.filename)) is not None:
```

Comment:
I was able to find documentation for `?!` (negative look-ahead), but I never used this. I'll ask a couple naive questions to make sure I understand how this works.
Surrounding `()` makes it a non-capturing group, is that right ?
So, the content of `match` is unchanged when the search *does* succeed (good), but we now reject file names that have more than one substring matching `\d{4}` ?

If the goal is to only match the file *name* (excluding the directory), a simpler fix would be
```suggestion
        if (match := re.search(r"\d{4}", self.ds.basename)) is not None:
```

Reply:
@neutrinoceros You are right. I wanted to just make this take the last match with 4 digits as it contains the filename. But I think what you are suggesting is also pretty good and will work.

Reply:
Ok, since this is a pretty different approach I'll open a separate PR then.